addons/mail/wizard/mail_compose_message.py

```diff
@@ -243,7 +243,7 @@ def send_mail(self, auto_commit=False):
                         batch_mails_sudo |= self.env['mail.mail'].sudo().create(mail_values)
                     else:
                         post_params = dict(
-                            message_type=wizard.message_type,
+                            message_type=wizard.message_type.no_test(),
```

Comment:
Why do you need this?

Reply:
Because the possible values for the wizard are a subset of the possible values. See the discussion at https://github.com/odoo/odoo/pull/54276#discussion_r452342738
I am trying to make this less "obscure"...

Reply:
It is less obscure, but now it feels really awkward :confused: 
What about discarding checks after the value has been hashed?
The values would become silent when put in a dict.

Reply:
> What about discarding checks after the value has been hashed?
> The values would become silent when put in a dict.

It would work in this particular case, but not in general. It will fail in this example if we call `get_value_from_wiz_or_model` with a value coming from the `wizard` model.
```python
class Wizard(models.Model)
    _name = 'wizard'
    selection = fields.Selection([('a', 'A'), ('b', 'B')])

class Model(models.Model)
    _name = 'model'
    selection = fields.Selection([('a', 'A'), ('b', 'B'), ('c', 'C')])

    def get_value_from_wiz_or_model(self, selection_value):
        if selection_value == 'c':  # 'c' is not a valid value for the wizard
            do_something()
```